zhaquirks/lidl/ts011f_plug.py

```diff
@@ -0,0 +1,127 @@
+"""LIDL TS011F plug."""
+
+import asyncio
+
+from zigpy.profiles import zha
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODEL,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya.ts011f_plug import Plug_3AC_4USB
+
+
+async def tuya_magic_spell_wand(dev, tries=100) -> None:
+    """Initialize device so that all endpoints become available."""
+    import inspect
+
+    basic_cluster = dev.endpoints[1].in_clusters[0]
+
+    # The magic spell is needed only once.
+    # TODO: Improve by doing this only once (successfully).
+
+    # Magic spell - part 1
+    attr_to_read = [4, 0, 1, 5, 7, 0xFFFE]
+    if "tries" in inspect.getfullargspec(basic_cluster.read_attributes)[0]:
+        await basic_cluster.read_attributes(attr_to_read, tries=tries)
+    else:
+        await basic_cluster.read_attributes(attr_to_read)
+
+    # Magic spell - part 2 (skipped - does not seem to be needed)
+    # attr_to_write={0xffde:13}
+    # basic_cluster.write_attributes(attr_to_write)
+
+    # Leave with rejoin - may need to be adjuste to work everywhere
+    # or require a minimum zigpy version
+    # This should have the device leave and rejoin immediately triggering
+    # the discovery of the endpoints that appear after the magic trick
+    await dev.zdo.request(0x0034, dev.ieee, 0x01, tries)
+
+    app = dev.application
+    # Delete the device from the database
+    app.listener_event("device_removed", dev)
+
+    # Delete the device from zigpy
+    app.devices.pop(dev.ieee, None)
+
+
+def tuya_magic_spell(dev, tries=100) -> None:
```

Comment:
Is 100 excessive?

Reply:
I use 100 when configuring sleepy devices - 20 wasn't enough - as far as I remember the "difficult" one responded after about 50 tries.  The cool thing about it was that I wasn't even on location when that happened.

I set 100 since @MattWestb wants to try it on sleepy devices, it is set to 3 for the plug.

The loop stops as soon as the packet is confirmed.

Reply:
For sleepy devices folks should be waking them up to pair them and keeping them awake (this is the guidance that is provided).

> The cool thing about it was that I wasn't even on location when that happened.

I don't think we want to encourage users to attempt to configure sleepy devices remotely. This falls into the just because you can doesn't mean you should camp IMHO

Side note: 
I think we are debating creating some sort of command queue for devices. The thought is we will queue up commands for devices that aren't reachable / are sleeping to be processed when they check in. This is probably a ways off for now though.

Reply:
Sleepers shall always being waked up and better implanting hits in the GUI for user doing it and trying sending somthing to Xiaomi sleepers is very welcome trying it.

Only thing i think can being good is implanting queue is for ZB3 sleepers that is having working pull control and making check ins (Z2M have doing it but not all the way) but can being good for some cases but real sleepers is not made for receiving commands only then being joined and then only sleeping ans sending little now and then.
"Half sleepers" (end device with radio on the idle) like TRVs and some switches can benefiting for queue if its good implanted but i dont  have so much experience of them.

Reply:
A command queue is definitively usefull, but I am not sure it will be sufficient for sleepy devices where the parent is a router.

I haven't digged in that part of ZB yet, but I am under the impression that the pending message has to be waiting at the router so that it can reply to the device when it checks in.  
I am not sure that it's relayed up to the coordinator before the device goes to sleep again.

One may need to act on a report that needs an ack so that the ack and the pending message is sent "at the same time".  Or, know the periodicity of the checkins and send the command just prior to that so that it is still waiting in the router.  + there is Green Power.

Reply:
If the coordinator is ZB3 and the SED and its parent and having pull control implanted chickins shall working OK but only if all is supporting pull control so no old CC-2431 coordinators or routers with HA zigbee stack.